From 4a95bdfcb7d9c8cfe2eb9521bcb20e27f28ab7c0 Mon Sep 17 00:00:00 2001
From: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
Date: Mon, 9 May 2022 15:43:48 +0800
Subject: [PATCH] opencv3: enable support for dnn library

opencv dnn library configuration is not enable in
the configuration. This patch enable opencv to
build with dnn library.

Signed-off-by: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
---
 package/opencv3/Config.in  |  7 +++++++
 package/opencv3/opencv3.mk | 10 +++++++++-
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/package/opencv3/Config.in b/package/opencv3/Config.in
index b9ea033f13..c01cfb16e6 100644
--- a/package/opencv3/Config.in
+++ b/package/opencv3/Config.in
@@ -298,6 +298,13 @@ comment "protobuf support needs a toolchain w/ gcc >= 4.8"
 	depends on BR2_PACKAGE_PROTOBUF_ARCH_SUPPORTS
 	depends on !BR2_TOOLCHAIN_GCC_AT_LEAST_4_8
 
+config BR2_PACKAGE_OPENCV3_DNN
+	bool "dnn support"
+	select BR2_PACKAGE_OPENCV3_WITH_PROTOBUF
+	select BR2_PACKAGE_PROTOBUF
+	help
+	  Use shared dnn library
+
 config BR2_PACKAGE_OPENCV3_WITH_TIFF
 	bool "tiff support"
 	select BR2_PACKAGE_TIFF
diff --git a/package/opencv3/opencv3.mk b/package/opencv3/opencv3.mk
index 5c82da9975..872c234762 100644
--- a/package/opencv3/opencv3.mk
+++ b/package/opencv3/opencv3.mk
@@ -108,7 +108,8 @@ OPENCV3_CONF_OPTS += \
 	-DBUILD_opencv_videoio=$(if $(BR2_PACKAGE_OPENCV3_LIB_VIDEOIO),ON,OFF) \
 	-DBUILD_opencv_videostab=$(if $(BR2_PACKAGE_OPENCV3_LIB_VIDEOSTAB),ON,OFF) \
 	-DBUILD_opencv_viz=OFF \
-	-DBUILD_opencv_world=OFF
+	-DBUILD_opencv_world=OFF \
+	-DBUILD_opencv_dnn=$(if $(BR2_PACKAGE_OPENCV3_LIB_DNN),ON,OFF)
 
 # Hardware support options.
 #
@@ -258,6 +259,13 @@ OPENCV3_CONF_OPTS += -DWITH_GTK=ON -DWITH_GTK_2_X=OFF
 OPENCV3_DEPENDENCIES += libgtk3
 endif
 
+ifeq ($(BR2_PACKAGE_OPENCV3_DNN),y)
+OPENCV3_CONF_OPTS += -DBUILD_opencv_dnn=ON \
+		     -DPROTOBUF_UPDATE_FILES=ON \
+		     -DWITH_PROTOBUF=ON
+OPENCV3_DEPENDENCIES += protobuf
+endif
+
 ifeq ($(BR2_PACKAGE_OPENCV3_WITH_JASPER),y)
 OPENCV3_CONF_OPTS += -DWITH_JASPER=ON
 OPENCV3_DEPENDENCIES += jasper
-- 
2.17.1

